### PR TITLE
fix(s3): allow accessLogsPrefix without accessLogsBucket

### LINKED ISSUE #6599: Bucket serverAccessLogsPrefix can't be defined independently from serverAccessLogsBucket
Here defined validation that `prefix` couldn't be specified without the `bucket`

https://github.com/aws/aws-cdk/blob/c9b074b28b26a537fb81b9a4a431df6412964c59/packages/%40aws-cdk/aws-s3/lib/bucket.ts#L1296

But in fact, [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-loggingconfig.html) don't say anything about this dependency and actually it works perfectly fine independently with the raw CF

### Reproduction Steps

```javascript
new s3.Bucket(scope, "Bucket", {serverAccessLogsPrefix: "pref"});
```

### Error Log

`Error: "serverAccessLogsBucket" is required if "serverAccessLogsPrefix" is set`

### Environment

  - **CLI Version      : *1.27.0 (build a98c0b3)*
  - **Framework Version: *1.27.0*
  - **OS               : *mac*
  - **Language         : *js*

### Other

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by lanwen**
fixes #6599

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by iliapolo** [comment]
@lanwen You're right the CFN docs say those properties are both optional. But I wonder, where will the log files be stored without defining the `serverAccessLogsBucket `?

**Comment by lanwen** [comment]
@iliapolo as stated in docs:
> By default, logs are stored in the bucket where the LoggingConfiguration property is defined.

File: packages/@aws-cdk/aws-s3/lib/bucket.ts [hunk 1]
```diff
@@ -1293,16 +1293,12 @@ export class Bucket extends BucketBase {
   }
 
   private parseServerAccessLogs(props: BucketProps): CfnBucket.LoggingConfigurationProperty | undefined {
-    if (props.serverAccessLogsPrefix && !props.serverAccessLogsBucket) {
-      throw new Error(`"serverAccessLogsBucket" is required if "serverAccessLogsPrefix" is set`);
-    }
-
-    if (!props.serverAccessLogsBucket) {
+    if (!props.serverAccessLogsBucket && !props.serverAccessLogsPrefix) {
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-s3/lib/bucket.ts]: This requires a change to the docstrings..the default behavior changed.

https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-s3/lib/bucket.ts#L907

**Comment by iliapolo** [comment]
Hi @lanwen 

Notice there are conflicts now, can you have a look? I can also take over if you'd like.

**Comment by lanwen** [comment]
@iliapolo Hey, rebased and updated, ready for merge :D

**Pull request merged**